cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionMetadata.java

```diff
@@ -68,14 +77,23 @@ public boolean equals(Object o) {
       return false;
     }
 
-    PartitionMetadata that = (PartitionMetadata) o;
+    PartitionMetadata entries = (PartitionMetadata) o;
```

Comment:
Seems like the `!=` in your suggestion should be a `==`. But otherwise, I updated my code.